pkg/evaluator/builtin.go

```diff
@@ -94,6 +95,38 @@ func hasFunc(args []Value) Value {
 	return &Bool{Val: ok}
 }
 
+var delDecl = &parser.FuncDecl{
+	Name: "del",
+	Params: []*parser.Var{
+		{Name: "map", T: parser.ANY_TYPE},
+		{Name: "key", T: parser.STRING_TYPE},
+	},
+	ReturnType: parser.NONE_TYPE,
+}
+
+func delFunc(args []Value) Value {
+	if len(args) != 2 {
+		return newError("'del' takes 2 argument not " + strconv.Itoa(len(args)))
+	}
+	m := args[0].(*Map)
+	keyStr := args[1].(*String)
+	key := keyStr.Val
+	order := make([]string, len(*m.Order)-1)
```

Comment:
by making a copy of order and replacing it, aren't you violating the reference semantics of maps? If you have:
```
a := {hello: "world"}
b := a
del a "hello"
```
then it should also be removed from b?

Thinking even more, I think b will be inconsistent - it will have `hello` in `Order`, but not in `Pairs` (because that is shared).

Instead I think you need to remove in-place and not allocate a new map - search for `key` in order and slice assign the element away - something like https://stackoverflow.com/a/37335777/23744 perhaps (although off the top of my head, I'm not 100% sure if that will not reallocate). It's a shame you can't use generics because x/slices has a delete function.

Reply:
Ok, I have addressed this finally. I've moved the delete functionality into the value.go alongside insertKey and Get. seems like the better place.